public/js/buttonFunctions.js

```diff
@@ -19,6 +19,29 @@ let newAssignment = function() {
   }
 }
 
+function replaceAssignmentFromID(oldData, newData, classID) {
+  const assignments = currentTableData.currentTermData.classes[classID].assignments;
+
+  //this weird version of indexOf is necessary (I think) because when comparing the two it doesn't always match correctly
+  let index = -1;
+  for (i=0; i<assignments.length-1; i++) {
+    if (assignments[i]["assignment_id"] === oldData["assignment_id"]) {
+      index = i;
+      break;
+    }
+  }
```

Comment:
I think the reason why using regular `indexOf` does not work here is because it expects perfect equality between objects and not just structural equality or equality of assignment IDs. In any case, this can be refactored in a way that avoids a C-style `for` loop and is more readable (using [ES6 features](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex)):
```suggestion
  const index = assignments.findIndex(({ assignment_id }) =>
    assignment_id === oldData.assignment_id
  );
```